### ISSUE TITLE
UX tweak suggestion: move "copy link" out of the ... menu that includes "block user"

### ISSUE THREAD

**Issue by strypey** [nolanlawson/pinafore#1670]
The other day I accidentally blocked a user while trying to copy a link to one of their posts. The ellipse menu under a post has 5 different options:
* Unfollow
* Block
* Mute
* Report
* Copy link

One of these things, as the sages of Sesame Street used to say, is not like the other. Would it be possible to move the 'copy link' somewhere else? Maybe a button next to, or underneath the UI element that contains the internal Pinafore link for that post?

**Comment by nolanlawson** [comment]
I think organizing these with sub-headers would make sense.

**Comment by strypey** [comment]
@nolan
> I think organizing these with sub-headers would make sense.

It's your party, but IMHO it's better UX to have one menu for moderation tasks, and entirely separate UI elements for things that are not moderation tasks. 'Copy link' is something I do frequently enough that I'd prefer not to have to open a burger menu to access that control. If it's in a burger menu that also contains controls that makes irreversible changes (like 'block'), that means I can't do it quickly without risking the kind of mistake I mentioned in the OP.

**Comment by Krinkle** [comment]
For what its worth, I think separate top-level menus or sub menus would clutter the interface, and require more overhead for both scenarios. Finding a meaningful and international icon and being friendly to new users would also be challenging.

However, I do want voice also the concern that a single menu for all items is indeed problematic and prone to error -if- the destructive actions have no confirmation behind them.

I've already lost several toots due to accidental deletions due to this via a different mastodon client I tried, after a mis-tap.

Adding indirection the menu seems undesirable as it means longer paths to discovery and more significant/distracting choices to make. However adding a confirmation to the end of it, I think, could strike a good balance here.

I notice that Mute account and Report toot already lead to modals.

Delete toot is currently direct.
Unfollow is also direct.

Mute thread and report account are also direct, but those are easy to undo, and don't produce visible side effects or notifications (as far as I know).

**Comment by Krinkle** [comment]
Somewhat related: Would it make sense to adopt a convention to indicate whether a link does something vs starts a modal?

Eg "copy link" directly replaces the user's clipboard. The others could have "..." after then per the desktop UI convention for menu items that have additional steps to them.

**Comment by strypey** [comment]
@Krinkle This is not a hill I plan to die on, but I think a menu with too many choices in it is also bad UX, especially if some of those choices are *very* different in nature from the others, as mentioned in my previous comment. IMHO a cluttered UI is best minimized by grouping controls that are alike in some way, under menus etc.

> Mute thread and report account are also direct, but those are easy to undo

Reporting an account to admins is *not* something I would want to do automatically. It could create considerable aggravation for the reported user, for admins, and for me. I would definitely like a confirmation modal on that one ;)

**Comment by Krinkle** [comment]
> > Mute thread and ~~report~~ account are also direct, but those are easy to undo
> 
> Reporting an account to admins is _not_ something I would want to do automatically. […]

Indeed, and it isn't. My bad. I meant to say "Mute thread and **block** account.". As I wrote earlier in the message, "Report" already leads to a modal workflow, and indeed does not and should not do anything automatically from the initial tap.
